subt_ign/launch/cave_circuit.ign

```diff
@@ -1006,38 +1086,57 @@
 %>
 
 <%
+
+  # Reverse hash of mersupials, i.e, key: child, value: parent
+  marsupialChildren = marsupialParents.invert
+
+
+  # Spawn nonmarsupial and parent marsupial robots first. Then spawn child
+  # marsupials since they need the positions of their parents
+  spawnList = robots.select { |name| not marsupialChildren.has_key?(name) }.values +
+              robots.select { |name| marsupialChildren.has_key?(name) }.values
+
   robotSpawned = 0
-  robots.each do |name, config|
-    robotType = config[1]
-    robotConfigN = config[-1]
+  spawnList.each do |robot|
+    name = robot.name
+    config = robot.config
+    robotType = robot.type
+    robotConfigN = robot.configNumber
     posX = -(robotSpawned / spawnColSize * spawnGridSize - spawnRowOffset)
     posY = -(robotSpawned % spawnColSize * spawnGridSize - spawnColOffset)
 
-    x = Math.cos(-spawnWorldYaw) * posX + Math.sin(-spawnWorldYaw) * posY
-    y = -(Math.sin(-spawnWorldYaw)) * posX + Math.cos(-spawnWorldYaw) * posY
-    posX = x + spawnWorldXPos
-    posY = y + spawnWorldYPos
+    # Spawn above parent position if this robot is marsupial child
+    if marsupialChildren.has_key?(name)
+      parentRobot = robots[marsupialChildren[name]]
+      robot.pos.x = parentRobot.pos.x + MARSUPIAL_PARENT_ROBOT_POSITION_OFFSETS[parentRobot.type].x
+      robot.pos.y = parentRobot.pos.y + MARSUPIAL_PARENT_ROBOT_POSITION_OFFSETS[parentRobot.type].y
```

Comment:
apply rotation to offset? e.g. 

```
      mx = MARSUPIAL_PARENT_ROBOT_POSITION_OFFSETS[parentRobot.type].x
      my = MARSUPIAL_PARENT_ROBOT_POSITION_OFFSETS[parentRobot.type].y
      robot.pos.x = parentRobot.pos.x +
          Math.cos(-spawnWorldYaw) * mx + Math.sin(-spawnWorldYaw) * my
      robot.pos.y = parentRobot.pos.y +
         -(Math.sin(-spawnWorldYaw)) * mx + Math.cos(-spawnWorldYaw) * my
```

 only in case we need to reuse this logic later but it shouldn't make a difference for cave circuit

Reply:
Done. d964ab719218723af0278ee03d3a2f2e70c9d88f